elastic/integrations
# [Stack Monitoring] update packages dataset

### Summary

Closes #3929

This change updates the datasets of the Stack Monitoring packages to include a stack_monitoring part (ie `elasticsearch.stack_monitoring.node`. The identifier has two purposes: 1) clarify the intent of these datastreams 2) free the stack products namespaces for the upcoming PO initiative. Only the metrics are impacted because logs will be collected similarly in both SM and PO.

The change also bumps elasticsearch, kibana and logstash package to their next major version while keeping their release to experimental until https://github.com/elastic/kibana/issues/120415 is completed. This will allow the packages to be built into the registry so that we don't have to manually build them locally when testing. I thought that metrics mappings being aligned was a relevant milestone for the bump.

### Testing
- Start an elastic-package stack with elasticsearch, kibana and logstash packages installed:
  - We can automate the package installation by providing the right fleet configuration to kibana, and we can use [elastic-package profiles](https://github.com/elastic/elastic-package#elastic-package-profiles) to do that. Let's download a profile that specifically does that. You can skip this step if you want to install the packages manually
  ```
  curl https://drive.google.com/uc\?export\=download\&id\=1aqdqNb9JaYXL-C3WjiT2t58Q65Cl4NIV -L -o /tmp/stack_monitoring-profile.zip && \
  unzip -o /tmp/stack_monitoring-profile.zip -d ~/.elastic-package/profiles && \
  rm /tmp/stack_monitoring-profile.zip
  ```
  - Now cd at the root of the integrations repository and let's build the packages, start the stack with the downloaded profile and also start a logstash service with some predefined pipelines. The command may take a moment to complete and will be done once the logstash service is started, this log message should appear: `Service is up, please use ctrl+c to take it down`
  ```
  (cd packages/elasticsearch && elastic-package build) && \
  (cd packages/kibana && elastic-package build) && \
  (
    cd packages/logstash && elastic-package build && \
    elastic-package stack up -v -d --profile stack_monitoring --version 8.5.0-SNAPSHOT && \
    elastic-package service up -v
  )
  ```
- Start a local Kibana with this change https://github.com/elastic/kibana/pull/137904. See [how to connect a local kibana](https://github.com/elastic/observability-dev/blob/548dfc0c5fe2776af40c1e85adc515c38139b4e5/docs/infra-obs-ui/stack-monitoring_integration-packages.md#connecting-a-local-kibana)
- Open kibana at `http://localhost:5602` and navigate to Stack Monitoring app. Elasticsearch, Kibana and Logstash section should be showing up with all their views populated. Inspect all the views and report anything bizarre
- Verify every stack monitoring `metrics-*` data streams is formatted as `metrics-{product}.stack_monitoring.{metricset}`
